### PR TITLE
fix(@rjsf/chakra-ui) Unify appearance of chakra-ui SelectWidget for single and multiple-choice

### LINKED ISSUE #3180: Unify appearance of chakra-ui SelectWidget for single and multiple-choice
### Prerequisites

- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)

### What theme are you using?

chakra-ui

### Is your feature request related to a problem? Please describe.

There are two minor UI issues that could be fixed:

1. Currently the chakra-ui `SelectWidget` uses `Select` from `chakra-react-select` for multi-select and `Select` from `@chakra-ui/react` for single-select.
As the result, styling (especially in dark mode) looks different:
image
image

2. In multi-select, once selected, the values are displayed rather than labels:
image
image

 

### Describe the solution you'd like

1. Use `Select` from `chakra-react-select` for both single- and multiple-choice;
2. Display label for selected items in multiple-choice `SelectWidget`.

If `@rjsf/chakra-ui` developers are okay with this kind of change, I'd like to contribute.

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by ldorofeeva**
### Reasons for making this change

fixes #3180 

Changes:
- In `SelectWidget`: use `Select` from `chakra-react-select` for both single- and multiple-choice select
- In `SelectWidget` multiple-choice select: display label rather than value for selected items

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/chakra-ui/src/SelectWidget/SelectWidget.tsx [hunk 1]
```diff
@@ -20,20 +16,13 @@ const SelectWidget = (props: WidgetProps) => {
     disabled,
     readonly,
     value,
-    autofocus,
     onChange,
-    onBlur,
-    onFocus,
```
Comment (heath-freenome) [inline packages/chakra-ui/src/SelectWidget/SelectWidget.tsx:26]: Why did you remove these three props and where they were used?

Comment (ldorofeeva) [reply packages/chakra-ui/src/SelectWidget/SelectWidget.tsx]: Good point,
They were only used for single-choice `Select` widget from chakra, but not for multiple-choice `Select` widget from `chakra-react-select` which made me think these props are not supported by latter. 

However `onFocus` and `onBlur` are ([as of January](https://github.com/csandman/chakra-react-select/commit/449d5b3bbc80ce14533f55b3c75402db185de844)), so I've added them back in [f5339b2](https://github.com/rjsf-team/react-jsonschema-form/pull/3182/commits/f5339b263e72d9865c7a397d4a5d19b000bf7fb4).

Comment (ldorofeeva) [reply packages/chakra-ui/src/SelectWidget/SelectWidget.tsx]: I've also added back the `autoFocus` prop as of [ea34f34](https://github.com/rjsf-team/react-jsonschema-form/pull/3182/commits/ea34f349af260cdb3377bc45d60a2cec8103dba5)

File: packages/chakra-ui/src/SelectWidget/SelectWidget.tsx [hunk 2]
```diff
@@ -78,42 +77,36 @@ const SelectWidget = (props: WidgetProps) => {
...
             };
           })}
         />
       ) : (
         <Select
-          id={id}
+          inputId={id}
           name={id}
-          placeholder={placeholder !== "" ? placeholder : " "}
-          value={typeof value === "undefined" ? emptyValue : value.toString()}
```
Comment (heath-freenome) [inline packages/chakra-ui/src/SelectWidget/SelectWidget.tsx:101]: I don't see where you are passing in the value for this one

File: packages/chakra-ui/src/SelectWidget/SelectWidget.tsx [hunk 3]
```diff
@@ -78,42 +77,36 @@ const SelectWidget = (props: WidgetProps) => {
         </FormLabel>
       )}
       {typeof multiple !== "undefined" && enumOptions ? (
```
Comment (heath-freenome) [inline packages/chakra-ui/src/SelectWidget/SelectWidget.tsx:79]: Might I suggest refactoring this to a `const isMultiple = multiple !== "undefined" && Boolean(multiple);` and then you can simplify the if/else into a single render... Something like:
```
const isMultiple = multiple !== "undefined" && Boolean(multiple);
let formValue;
if (isMultiple) {
  formValue = (value || []).map((v: any) => ({
    label: _valueLabelMap[v],
    value: v,
  }));
} else {
  formValue = value || "";
}
...
       <Select
          inputId={id}
          name={id}
          isMulti={isMultiple}
          options={enumOptions as OptionsOrGroups<unknown, GroupBase<unknown>>}
          placeholder={placeholder}
          closeMenuOnSelect={!isMultiple}
          onBlur={_onBlur}
          onChange={isMultiple ? _onMultiChange : _onChange}
          onFocus={_onFocus}
          autoFocus={autofocus}
          value={formValue}
        />
...
```

**Review by heath-freenome — CHANGES_REQUESTED** [review]

**Comment by heath-freenome** [comment]
@ldorofeeva It looks like you have to run the `npm run cs-format` command. FYI, if you do `npm run prepare` from the root directory, it will set up `husky` and it will run the `pre-commit:husky` script for you to automatically do this for your files

**Comment by ldorofeeva** [comment]
> FYI, if you do `npm run prepare` from the root directory, it will set up `husky` and it set up a precommit hook for git that will run the `pre-commit:husky` script for you to automatically do this for your files

Awesome! Thanks for the advise! Haven't seen it in the contributor's guide

**Comment by heath-freenome** [comment]
@ldorofeeva When you are ready, do move this out of Draft and rebase to main

**Review by heath-freenome — APPROVED** [review]
